tools/reviewer/job/sync/GitHubClient.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright 2018 The StartupOS Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.startupos.tools.reviewer.job.sync;
+
+import com.google.protobuf.util.JsonFormat;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.UserRequest;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.UserResponse;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.PullRequestRequest;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.PullRequestResponse;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.CommentsRequest;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.CommentsResponse;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.PullRequestFilesRequest;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.PullRequestFilesResponse;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.CreatePullRequestResponse;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.CreatePullRequestCommentRequest;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.CreatePullRequestRequest;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.CreateReviewCommentRequest;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.PullRequestsResponse;
+import com.google.startupos.tools.reviewer.job.sync.GitHubProtos.PullRequestsRequest;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Base64;
+
+import static java.net.HttpURLConnection.HTTP_CREATED;
+import static java.net.HttpURLConnection.HTTP_OK;
+
+public class GitHubClient {
+  private static final String BASE_PATH = "https://api.github.com/";
+  private static final String GET_SINGLE_PULL_REQUEST = "repos/%s/pulls/%d";
```

Comment:
Can you add links to GitHub's documentation for every request? E.g this one would be: https://developer.github.com/v3/pulls/#get-a-single-pull-request

Reply:
Done.